utils/data.py

```
@@ -0,0 +1,46 @@
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Corrupted JSON file: {path}",
+        )
+
+    except OSError as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to read JSON file: {path} ({e})",
+        )
+
+
+def write_json(path: Path, data: Any) -> None:
```

Ruff (0.14.11)

Dynamically typed expressions (typing.Any) are disallowed in `data`

(ANN401)